bootstrap/api/v1beta2/kthreesconfig_types.go

```diff
@@ -0,0 +1,271 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KThreesConfigSpec defines the desired state of KThreesConfig.
+type KThreesConfigSpec struct {
+	// Files specifies extra files to be passed to user_data upon creation.
+	// +optional
+	Files []File `json:"files,omitempty"`
+
+	// PreK3sCommands specifies extra commands to run before k3s setup runs
+	// +optional
+	PreK3sCommands []string `json:"preK3sCommands,omitempty"`
+
+	// PostK3sCommands specifies extra commands to run after k3s setup runs
+	// +optional
+	PostK3sCommands []string `json:"postK3sCommands,omitempty"`
+
+	// AgentConfig specifies configuration for the agent nodes
+	// +optional
+	AgentConfig KThreesAgentConfig `json:"agentConfig,omitempty"`
+
+	// ServerConfig specifies configuration for the agent nodes
+	// +optional
+	ServerConfig KThreesServerConfig `json:"serverConfig,omitempty"`
+
+	// Version specifies the k3s version
+	// +optional
+	Version string `json:"version,omitempty"`
+}
+
+// TODO
```

Comment:
I don't think so. Was added as part of https://github.com/k3s-io/cluster-api-k3s/commit/73fb3f035d55e0dc760539d94c60ec25071448c0 and it may still be relevant.